replace_with/src/lib.rs

```diff
@@ -0,0 +1,10 @@
+use std::ptr;
+
+pub fn replace_with<T, F>(t: &mut T, f: F) where F: FnOnce(T) -> T
+{
+	unsafe
+	{
+		let t_ptr = t as *mut T;
+		ptr::write(t, f(ptr::read(t_ptr)));
```

Comment:
lol I have no idea if this would work or not. I re-committed to replace t with t_ptr, but RUST wasn't giving me a type error sooooooo....